homeassistant/components/recorder/history/__init__.py

```diff
@@ -92,7 +96,15 @@ def get_significant_states(
     minimal_response: bool = False,
     no_attributes: bool = False,
     compressed_state_format: bool = False,
-) -> MutableMapping[str, list[State | dict[str, Any]]]:
+) -> MutableMapping[
+    str,
+    list[
+        State
+        | HistoryCompressedState
+        | HistoryMinimalCompressedState
+        | HistoryMinimalState
+    ],
```

Comment:
The return typing is not correct, it should be:
```py
MutableMapping[
    str,
    list[State | HistoryMinimalState]
    | list[HistoryCompressedState | HistoryMinimalCompressedState],
]:
```

Or maybe:
MutableMapping[
    str,
    list[State]
    | list[State | HistoryMinimalState]
    | list[HistoryCompressedState],
    | list[HistoryCompressedState | HistoryMinimalCompressedState],
]:
```